CMSgov/mint-app
# [Easi-4043] User notification preferences flag to array

# EASI-4043


## Changes and Description

- Migrated user notification preferences to use an array instead of a single enum value
- introduced a serialization package to host helper methods to serialize this enum array (originally prototyped in `EASI-1982/generic_type_array_serialization`
   - We could have used a pq string array, but it seems important to be able to use receiver methods on this preference type. We will be using this value throughout the application.

## How to test this change
1. Use the updated postman collection to update user preferences.
2. Try to remove all preferences, and re-add preferences.
3. Verify the existing logic to not send an email if tagged in a discussion or a reply continues to function as expected.


Note: This code does not enforce that a preference not be duplicated in an entry, for example, IN_APP could be listed twice. The application will still handle this, considering the presence of the enum at least once to be enough to send a notification. 



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
